Dynatrace/dynatrace-operator
# EdgeConnect status of CR doesnt show real situation, when pod is failing

## Description

THis PR is bugfix for https://dt-rnd.atlassian.net/browse/K8S-11360 

Before this PR:

```
kaf ../samples/edgeconnect_****_v1alpha1.yaml
edgeconnect.dynatrace.com/e2e-andrii created
secret/edge-connect-client-secret-***** unchanged
```

```
➜  $ k get ec
NAME         APISERVER                             STATUS    AGE
e2e-andrii   *****.dev.apps.dynatracelabs.com   Running   5s
```

but pod in crashloop state.

After this change status will be Deploying, because we are not reached desired amount of replicas.

## How can this be tested?

Deploy different edgeConnects, especially broken ones.

Make sure CR status represents the actual state of EdgeCOnnect Deployment.
